server/src/Repositories/ProjectsRepository.cs

```diff
@@ -79,6 +79,31 @@ public Project InsertProject(SqlTransaction transaction, Db db, string projectNa
             }
         }
 
+        public Project UpdateProject(Db db, int projectId, string projectName, string Repo, string Token, bool PublicProject)
+        {
+            string sql = @"UPDATE [Project] SET ProjectName=@ProjectName, Repo=@Repo, Token=@Token, PublicProject=@PublicProject OUTPUT INSERTED.ProjectID, INSERTED.ProjectName, INSERTED.Repo, INSERTED.Token, INSERTED.PublicProject WHERE ProjectID=@ProjectID;";
+
+            using (SqlCommand command = new(sql, db.Connection))
+            {
+                command.Parameters.Add("@ProjectID", System.Data.SqlDbType.Int).Value = projectId;
+                command.Parameters.Add("@ProjectName", System.Data.SqlDbType.VarChar).Value = projectName;
+                command.Parameters.Add("@Repo", System.Data.SqlDbType.VarChar).Value = Repo;
+                command.Parameters.Add("@Token", System.Data.SqlDbType.VarChar).Value = Token;
+                command.Parameters.Add("@PublicProject", System.Data.SqlDbType.Bit).Value = PublicProject;
+
+                using SqlDataReader reader = db.ExecuteReader(command);
+
+                if (reader.Read())
+                {
+                    return new Project(reader.GetInt32("ProjectID"), reader.GetString("ProjectName"), reader.GetString("Repo"), reader.GetString("Token"), reader.GetBoolean("PublicProject"));
+                }
+                else
```

Comment:
No need for this because the OUTPUT is guaranteed to exist. So reader.Read() does not need to be in an if statement.

Reply:
If the update fails for some reason this picks it up